libraries/AP_Arming/AP_Arming.cpp

```diff
@@ -493,8 +493,16 @@ bool AP_Arming::arm_checks(uint8_t method)
 {
     if ((checks_to_perform & ARMING_CHECK_ALL) ||
         (checks_to_perform & ARMING_CHECK_LOGGING)) {
-        if (!DataFlash_Class::instance()->logging_started()) {
+        // allow logging to start by indicating (falsely) that the vehicle is armed:
+        DataFlash_Class *df = DataFlash_Class::instance();
+        df->set_vehicle_armed(true);
```

Comment:
I understand the reason but I don't really like this. What if another arm check fails? It will continue writing to log.